raiden/network/proxies/token_network.py

```diff
@@ -162,6 +163,17 @@ def _call_and_check_result(
 
         return call_result
 
+    def lock_and_execute_channel_operation(self,
+                                           channel_identifier: ChannelID,
+                                           semaphore: Optional[Semaphore],
+                                           code_to_block) -> Optional[Any]:
+        if semaphore:
+            with self.channel_operations_lock[channel_identifier], semaphore:
+                return code_to_block()
+        else:
+            with self.channel_operations_lock[channel_identifier]:
+                return code_to_block()
```

Comment:
i don't like this `if` so much, but i don't see a very elegant alternative; i can only think of making `semaphore` non optional, and then passing a new `RLock` every time the function is called from outside in the cases where originally `semaphore is None` (this new `RLock` would always be free to use).

i say leave as is

Reply:
ok scratch that, i have an idea.

what if we remove `semaphore` from the params, and then do this:

```python
def lock_and_execute_channel_operation(self, channel_identifier: ChannelID, code_to_block) -> Optional[Any]:
        with self.channel_operations_lock[channel_identifier]:
            return code_to_block()
```

and then:

```python
def make_deposit():
    with self.deposit_lock:
        # rest of code
```

Reply:
Got it i will try it